app/validator.js

```diff
@@ -27,6 +27,14 @@ const customJoi = Joi.extend((joi) => ({
     ]
 }));
 
+const vizIdPrefix='threshold_alert_:';
+
+function preprocessor(result){
+	result.value.report_line_item.visualizationid=result.value.report.thresholdAlert?vizIdPrefix+result.value.report_line_item.visualizationid:result.value.report_line_item.visualizationid;
```

Comment:
it's incredibly hard to read lines with zero space and that are such long. Maybe it's better to have something like:

```
result.value.report_line_item.visualizationid=result.value.report.thresholdAlert?vizIdPrefix+result.value.report_line_item.visualizationid:result.value.report_line_item.visualizationid;
 
 
it's incredibly hard to read lines with zero space and that are such long. Maybe it's better to have something like:

```
result.value.report_line_item.visualizationid = result.value.report.thresholdAlert 
? vizIdPrefix + result.value.report_line_item.visualizationid
: result.value.report_line_item.visualizationid;
```